llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -195,6 +195,14 @@ static bool hasUndefinedPassthru(const MachineInstr &MI) {
   return UseMO.getReg() == RISCV::NoRegister || UseMO.isUndef();
 }
 
+/// Return true if \p MI is a copy that will be lowered to one or more vmvNr.vs.
+static bool isVectorCopy(const TargetRegisterInfo *TRI,
+                         const MachineInstr &MI) {
+  return MI.isCopy() && MI.getOperand(0).getReg().isPhysical() &&
```

Comment:
Where do we lower from the segmented load/store register classes?  I know those survive through vector register allocation (as in we have spill fill supoort), do we need to handle them here?

Reply:
For copies that would happen in RISCVInstrInfo::copyPhysReg which is called from PostRA pseudo expansion. That would be after scalar regalloc.

Reply:
From what I understood they also take part in vector register allocation? The tuple register classes have the `IsVRegClass` TSFlag set, which is what `isRVVRegClass` checks for. I think the copy in `vlsegff-rv64.ll` is a segmented tuple, and it seems to be picked up in this PR. cc @4vtomat